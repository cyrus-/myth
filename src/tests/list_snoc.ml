let s = "
type nat =
  | O
  | S of nat

type list =
  | Nil
  | Cons of nat * list

let list_snoc : list -> nat -> list |>
  { [] => ( 0 => [0]
          | 1 => [1] )
  | [0] => ( 0 => [0; 0]
           | 1 => [0; 1] )
  | [1; 0] => ( 0 => [1; 0; 0]
              | 1 => [1; 0; 1] )
  | [2; 1; 0] => ( 0 => [2; 1; 0; 0]
                 | 1 => [2; 1; 0; 1] )
  } = ?"
