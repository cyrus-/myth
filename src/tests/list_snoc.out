(* Generated Thu Nov 13 00:35:51 2014 *)
let list_snoc : list -> nat -> list =
  let rec f1 (l1:list) : nat -> list =
    fun (n1:nat) ->
      match l1 with
        | Nil -> Cons (n1, Nil)
        | Cons (n2, l2) -> Cons (n2, f1 l2 n1)
  in
    f1
;;
